packages/block-library/src/list/edit.js

```diff
@@ -13,40 +25,104 @@ export default function ListEdit( {
 	onReplace,
 	className,
 } ) {
-	const { ordered, values } = attributes;
+	const { ordered, values, reversed, start, type, description } = attributes;
+
+	const listTypes = [
+		{
+			name: __( 'Decimal' ),
+			type: '1',
+			description: 'list-type-decimal',
+		},
+		{
+			name: __( 'Lower alpha' ),
+			type: 'a',
+			description: 'list-type-lower-alpha',
+		},
+		{
+			name: __( 'Upper alpha' ),
+			type: 'A',
+			description: 'list-type-upper-alpha',
+		},
+		{
+			name: __( 'Lower roman' ),
+			type: 'i',
+			description: 'list-type-lower-roman',
+		},
+		{
+			name: __( 'Upper roman' ),
+			type: 'I',
+			description: 'list-type-upper-roman',
+		},
+	];
 
 	return (
-		<RichText
-			identifier="values"
-			multiline="li"
-			tagName={ ordered ? 'ol' : 'ul' }
-			onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
-			value={ values }
-			wrapperClassName="block-library-list"
-			className={ className }
-			placeholder={ __( 'Write list…' ) }
-			onMerge={ mergeBlocks }
-			unstableOnSplit={
-				insertBlocksAfter ?
-					( before, after, ...blocks ) => {
-						if ( ! blocks.length ) {
-							blocks.push( createBlock( 'core/paragraph' ) );
-						}
+		<Fragment>
+			<RichText
+				identifier="values"
+				multiline="li"
+				tagName={ ordered ? 'ol' : 'ul' }
+				onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
+				value={ values }
+				start={ start }
+				reversed={ reversed }
+				type={ type }
+				wrapperClassName="block-library-list"
+				className={ ordered ? classnames( className, description ) : className }
```

Comment:
Hopefully, by removing the redundant CSS rule, the class name can also be removed. :)

Reply:
Done